app/sources/HFTclTemplateController.m

```diff
@@ -467,6 +471,46 @@ - (int)runCommand:(enum command)command objc:(int)objc objv:(struct Tcl_Obj * CO
             }
             break;
         }
+        case command_hfversion: {
+            if (objc != 4) {
+                Tcl_WrongNumArgs(_interp, 0, objv, "major minor patch");
+                return TCL_ERROR;
+            }
+            long major;
+            int err = Tcl_GetLongFromObj(_interp, objv[1], &major);
+            if (err != TCL_OK) {
+                return err;
+            }
+            long minor;
+            err = Tcl_GetLongFromObj(_interp, objv[2], &minor);
+            if (err != TCL_OK) {
+                return err;
+            }
+            long patch;
+            err = Tcl_GetLongFromObj(_interp, objv[3], &patch);
+            if (err != TCL_OK) {
+                return err;
+            }
+            /*
+                In order for this check to work:
```

Comment:
better to have an assert (runtime is fine) in the app.

but even better is to not have these requirements at all. doesn't look like Cocoa or Swift have any built-in version comparison, which is a bummer. but, we can make similar assumption about version string being 1-3 values, and just compare them each.

Reply:
I've changed up the code here quite a bit. Please have a look and let me know what you think.

Reply:
As for the logic - by smashing the three values into a single integer and then comparing them, it's a much more straightforward way of checking version numbers than looking at major/minor/patch individually, and seeing if they check out (which they can't - they can only be compared in relation to one another.)